test/smartwallet/smartWallet.test.ts

```diff
@@ -217,14 +445,66 @@ describe('SmartWallet contract', function () {
         .relayedUserSmartWalletCreation(
           deployRequest.request,
           suffixData,
-          owner.address,
+          worker.address,
           signature
         );
 
       expect(fakeToken.transfer).to.be.called;
     });
 
-    it('Should not call transfer on sponsored deployment', async function () {
+    it('Should pay on not sponsored(native) deployment', async function () {
+      const amountToBePaid = hardhat.utils.parseEther('0.01');
+      const deployRequest = createDeployRequest({
+        relayHub: relayHub.address,
+        from: owner.address,
+        nonce: '0',
+        tokenGas: '5000',
+        tokenAmount: amountToBePaid.toString(),
+        tokenContract: ZERO_ADDRESS,
+      });
+
+      const typedDeployData = new TypedDeployRequestData(
+        HARDHAT_CHAIN_ID,
+        smartWalletFactory.address,
+        deployRequest
+      );
+
+      const suffixData = getSuffixData(typedDeployData);
+
+      const privateKey = Buffer.from(owner.privateKey.substring(2, 66), 'hex');
+      const signature = getLocalEip712DeploySignature(
+        typedDeployData,
+        privateKey
+      );
+
+      const smartWalletAddress = await smartWalletFactory.getSmartWalletAddress(
+        owner.address,
+        ZERO_ADDRESS,
+        0
+      );
+      await owner.sendTransaction({
+        to: smartWalletAddress,
+        value: amountToBePaid,
+      });
+
+      const balanceBefore = await provider.getBalance(smartWalletAddress);
+
+      await smartWalletFactory
+        .connect(relayHub)
+        .relayedUserSmartWalletCreation(
+          deployRequest.request,
+          suffixData,
+          worker.address,
+          signature
+        );
+
+      const balanceAfter = await provider.getBalance(smartWalletAddress);
+
+      expect(balanceBefore).to.be.equal(amountToBePaid);
+      expect(balanceAfter).to.be.equal(balanceBefore.sub(amountToBePaid));
```

Comment:
I'd check for the receiver balance rather than the SW one

Reply:
updated